### ISSUE TITLE
Unidentified variable not being flagged as an error

### ISSUE THREAD

**Issue by ryancole** [eslint/eslint#7988]
**Tell us about your environment**

* **ESLint Version:** 3.12.2
* **Node Version:** 7.4.0
* **npm Version:** 3.10.10

**What parser (default, Babel-ESLint, etc.) are you using?**

babel-eslint v 7.1.1

**Please show your full configuration:**

```js
{
  "parser": "babel-eslint",
  "parserOptions": {
    "sourceType": "module",
    "ecmaVersion": 6,
    "ecmaFeatures": {
      "jsx": true,
      "impliedStrict": true,
      "experimentalObjectRestSpread": true
    }
  },
  "env": {
    "es6": true,
    "node": true,
    "browser": true
  },
  "plugins": [
    "react"
  ],
  "extends": [
    "eslint:recommended",
    "plugin:react/recommended"
  ],
  "rules": {
    "quotes": ["error", "double"],
    "no-console": "off"
  }
}
```

**What did you do? Please include the actual source code causing the issue.**

```js
import React from "react";
import ReactDOM from "react-dom";
import { match } from "@ryancole/router";
import { AppContainer } from "react-hot-loader";
import { ApolloProvider } from "react-apollo";

// the application's route handlers
import routes from "../../routes";

// our apollo graphql client
import graphqlClient from "./graphqlClient";

// given a location change, render route if needed
export default (async function renderLocation(location) {
  const route = match(location, routes);
  if (route) {
    const Component = await route.loadComponent(route, history);
    ReactDOM.render(
      <ApolloProvider client={graphqlClient}>
        <AppContainer>
          <Component />
        </AppContainer>
      </ApolloProvider>,
      document.getElementById("app")
    );
  }
});
```

**What did you expect to happen?**

I expected `history` to be identified as an unidentified variable. You can see that I use `history` but it's not defined anywhere. Sometimes this works with eslint, but in this case it's not being flagged as an error.

**What actually happened? Please include the actual, raw output from ESLint.**

I got no eslint error.

**Comment by not-an-aardvark** [comment]
Thanks for the report.

I think this is working as intended -- you have `env: { browser: true }` set in your config, and [`history`](https://developer.mozilla.org/en-US/docs/Web/API/Window/history) is a global variable in browsers.

**Comment by ryancole** [comment]
Ah true, good point. That's likely the reason!
